apis/database/v1beta1/cloudsql_instance_types.go

```diff
@@ -408,6 +412,86 @@ type OnPremisesConfiguration struct {
 	HostPort string `json:"hostPort"`
 }
 
+// ReplicaConfiguration Read-replica configuration for connecting to the primary instance.
+type ReplicaConfiguration struct {
+	// FailoverTarget: Specifies if the replica is the failover target. If
+	// the field is set to *true* the replica will be designated as a
+	// failover replica.
+	// +optional
+	FailoverTarget *bool `json:"failoverTarget,omitempty"`
+
+	// MysqlReplicaConfiguration: MySQL specific configuration when
+	// replicating from a MySQL on-premises primary instance. Replication
+	// configuration information such as the username, password,
+	// certificates, and keys are not stored in the instance metadata. The
+	// configuration information is used only to set up the replication
+	// connection and is stored by MySQL in a file named **master.info** in
+	// the data directory.
+	// +optional
+	MysqlReplicaConfiguration *MySqlReplicaConfiguration `json:"mysqlReplicaConfiguration,omitempty"`
+}
+
+// MySqlReplicaConfiguration: Read-replica configuration specific to
+// MySQL databases. https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1/instances#ReplicaConfiguration
+type MySqlReplicaConfiguration struct {
+
+	// SecretRef: Kubernetes Secret containing all credentials for MySqlReplicaConfiguration
+	// +optional
+	SecretRef *xpv1.SecretReference `json:"secretRef,omitempty"`
```

Comment:
Definitely that's great.
But, I'm wondering about controller implementation.

This will end up checking each `secretRef` provided and fetch it accordingly,
thinking about, in each reconciliation loop, the controller will fetch those secrets as user provided.
Isn't a bit noisy... (for controller perspective, as this will be implemented in the `Observe()`

Reply:
Yeah, that's a good point.

However, controller runtime's caching would help with this and we won't (really) fetch those secrets (or that secret if all put into 1 with different keys) in each reconcile from API Server.